app/models/email.rb

```diff
@@ -76,27 +76,54 @@ def self.weekly_skills_maintenance(organisation = nil)
                     end
     proficiencies.map do |proficiency|
       proficiency.users.map do |user|
-        if user.club.is_active && user.club.enable_reminders_email && user.club.show_skills_maintenance
+        if user.club.with_reminder_emails_enabled && user.club.show_skills_maintenance
           SwapseaMailer.proficiency_reminder(user, proficiency).deliver
           emails_sent += 1
         else
           Rails.logger.warn "Skipped sending skills maintenance email because #{user.club.name} is_active=#{user.club.is_active} and enable_reminders_email=#{user.club.enable_reminders_email} and show_skills_maintenance=#{user.club.show_skills_maintenance}"
         end
       rescue Exception => e
-        Rails.logger.warn "Skipped sending skills maintenance email because user:#{user.id} has no email."
+        Rails.logger.warn "Skipped sending skills maintenance email because user:#{user.id} has no email. #{e}"
       end
     end
-    "Sent #{emails_sent} skills maintenance emails."
+    "Sent #{emails_sent} skills maintenance emails for #{organisation}."
   end
 
-  def self.north_bondi_not_yet_proficient
-    ProficiencySignup.unsigned.each do |user_id|
-      user = User.find(user_id)
-      if user.club.is_active?
-        SwapseaMailer.north_bondi_not_yet_proficient(user).deliver
-      else
-        Rails.logger.warn "Skipped sending not yet proficient email because #{user.club.name} is_active=#{user.club.is_active}"
+  # Nudge open requestors to make an offer
+  def self.weekly_nudge_offers(organisation = nil)
+    clubs = if organisation
+              # Look up club id
+              Club.with_reminder_emails_enabled.where(name: organisation)
+            else
+              Club.with_reminder_emails_enabled
+            end
+
+    clubs.map do |club|
+      emails_sent = 0
+      # All open requests.
+      open_requests = Request.with_club(club.id).with_open_status
+      open_requests.map do |open_request|
+        # Open requests not by this user or for this roster.
+        other_requests = Request.with_club(club.id).with_open_status
```

Comment:
Persistence of the other_requests array? Maybe better to create new array, and append to it during loop.

Reply:
Would also need to remove possible duplicates. Possibly a better way?